### PR TITLE
ddl: View can be renamed only in the same schema

### LINKED ISSUE #32587: TiDB Rename view with different schema behavior is not compatable with mysql.
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```sql
drop database if exists  mysqltest;
drop table if exists t1;
drop view if exists v1;
create table t1 (a int);
create view v1 as select a from t1;
create database mysqltest;
rename table v1 to mysqltest.v1;
```
### 2. What did you expect to see? (Required)
```sql
mysql> rename table v1 to mysqltest.v1;
ERROR 1450 (HY000): Changing schema from 'test' to 'mysqltest' is not allowed.
```
### 3. What did you see instead (Required)
```sql
mysql> rename table v1 to mysqltest.v1;
Query OK, 0 rows affected (0.00 sec)

-- tidb is moving view v1  from existing schema to mysqltest schema .
```
### 4. What is your TiDB version? (Required)

```sql
| Release Version: v5.5.0-alpha-182-gad9430039
Edition: Community
Git Commit Hash: ad9430039f54bb9af78d44831c176bc5eafcbba0
Git Branch: master
UTC Build Time: 2022-02-18 17:57:13
GoVersion: go1.17.2
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false |
```

### PR DESCRIPTION
**Pull request by likzn**
### What problem does this PR solve?

Issue Number: close #32587

Problem Summary:

[Morgo](https://github.com/morgo) points out:
> The reason that I suspect MySQL does this, is that if the view definition is referencing t1, it is not clear if that is from the original schema or the new schema.
So MySQL avoids the ambiguity and returns an error. This is a much safer behavior than allowing the rename.

So when `extractTblInfos` check some `tableExists` and others , check if `rename view` of the schema is the same together.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Previously, it will be allowed when TiDB rename view table to different schema.But Mysql will forbid this operation. 
From now on,It will report error when TiDB rename view table to different schema.It is compatible with Mysql.
```

### CODE REVIEW

**Review by xhebox — APPROVED** [review]

**Review by xhebox — CHANGES_REQUESTED** [review]
@likzn Remember to fill release notes

Also, fix check_dev test plz

**Comment by likzn** [comment]
> @likzn Remember to fill release notes
> 
> Also, fix check_dev test plz

@xhebox The release note is added.
The problem of check_dev test. I found in https://github.com/pingcap/tidb/issues/27298#issue-972402667 . It seem like a historial problem. I will help to fix it in that issue.

**Review by xhebox — APPROVED** [review]

**Review by morgo — APPROVED** [review]
LGTM, I can't explain the CI failure. I've pulled in master, lets see if it fixes it.

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -5476,6 +5476,13 @@ func extractTblInfos(is infoschema.InfoSchema, oldIdent, newIdent ast.Ident, isA
 		// oldIdent is equal to newIdent, do nothing
 		return nil, 0, nil
 	}
+	//View can be renamed only in the same schema. Compatible with mysql
+	if is.TableIsView(oldIdent.Schema, oldIdent.Name) {
+		if oldIdent.Schema != newIdent.Schema {
```
Comment (hawkingrei) [inline ddl/ddl_api.go:5481]:
```
if is.TableIsView(oldIdent.Schema, oldIdent.Name) && oldIdent.Schema != newIdent.Schema {
```

Comment (likzn) [reply ddl/ddl_api.go:5481]: I think maybe it is more semantics to seprate two `if statement`
1.We judege the table is view or not firstly. And If it is, we continue to do other operations.
2.If there are other operations about `view table` in the future. We can add code directly below `if is.TableIsView(oldIdent.Schema, oldIdent.Name) {`

**Pull request merged**